### DISCUSSION TITLE
[General] zkPorter proving

### DISCUSSION THREAD

**Discussion by ckai0011** [zkSync-Community-Hub/zksync-developers#544] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Hi - a few questions on zkPorter:

ONE: 
per [this article](https://blog.matter-labs.io/zkporter-composable-scalability-in-l2-beyond-zkrollup-2a30c4d69a75)...

- [ ] does each state transition in a shard need a zkProof?  
- [ ] or, can the shard validators attest to the state transition and simply update the merkle root for that shard without a zk-proof?  

Here is the reference in that article:
"A shard’s smart contract defines its own rules for when a zkPorter block is accepted. When the block is accepted, the block enacts a state transition: a transaction on L1 that replaces the Merkle root with the new value. This transaction must verify a zero- knowledge proof that the new value is valid: i.e. that each of the transactions in the block was valid."

TWO: 
- [ ] if a zkProof is required, can disparate shards from the same project or different projects use the same prover?  Are there prover pools?  If so, does zkSync run this shared prover?

THREE: 
- [ ] can a shard have both on-chain and off-chain data availability at the same time?

Thank you very much.

**Comment by bxpana** [comment, marked as answer]
Right now the design of zkPorter has not been completed yet so since that article is so old, I would assume things would be slightly updated since then when zkPorter is launched. We'll probably see validiums on ZK Stack first and that will give you a good idea for how the system will work. I know that Lens is also planning on a Volition style chain like zkPorter so we'll see how that is designed.
